SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -21,15 +21,22 @@ def get(self):
         repo_type (str): The type of repository (git or eos).
         builder (str): The builder used for creating the environment.
         builder_version (str): The version of the specified builder.
+        nxcals (bool): Whether to include Nxcals in the environment.
         """
         self.set_header("Content-Type", "text/event-stream")
 
         repository = self.get_query_argument("repo", default="")
         repo_type = self.get_query_argument("repo_type", default="")
         builder = self.get_query_argument("builder", default="")
         builder_version = self.get_query_argument("builder_version", default="")
+        nxcals = self.get_query_argument("nxcals", default="")
+
+        arguments = ["--repo", repository, "--repo_type", repo_type, "--builder", builder]
+        if builder_version:
+            arguments.extend(("--builder_version", builder_version))
```

Comment:
This check for the builder version is not strictly related to NXCALS changes. Needs to be in a separate commit.